### PR TITLE
Scan: Implement query Band supply

### PR DESCRIPTION
**Pull request by Pitchanai**
Fixed: #2717 

## Implementation details
- Implement query band supply

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: scan/src/rpc/PriceHook.re [hunk 1]
```diff
@@ -20,23 +20,32 @@ module CoinGekco = {
...
 
     let data = {
       let%Opt usd = usdJson;
       let%Opt btc = btcJson;
-      Some(decode(usd, btc));
+      let%Opt supply =
+        switch (supplyJson) {
+        | None => None
+        | Some(sp) => sp |> Js.Json.decodeNumber
+        };
```
Comment (evilpeach) [inline scan/src/rpc/PriceHook.re]: Let's use `Belt.Option.map` like this
```
let%Opt supply = supplyJson->Belt.Option.map(sp |> Js.Json.decodeNumber);
```

File: scan/src/rpc/PriceHook.re [hunk 2]
```diff
@@ -46,22 +55,22 @@ module CoinGekco = {
 module CrytoCompare = {
   let get = () => {
     // TODO: Find the formular to calulate this
```
Comment (evilpeach) [inline scan/src/rpc/PriceHook.re]: remove `TODO`

File: scan/src/rpc/PriceHook.re [hunk 3]
```diff
@@ -46,22 +55,22 @@ module CoinGekco = {
...
-    let circulatingSupply = 20494032.;
-    let decode = (usdJson, btcJson) =>
+    let decode = (usdJson, btcJson, supplyJson) =>
```
Comment (evilpeach) [inline scan/src/rpc/PriceHook.re]: change `supplyJson` to `circulatingSupply `

File: scan/src/rpc/PriceHook.re [hunk 4]
```diff
@@ -70,19 +79,27 @@ module CrytoCompare = {
...
     let data = {
       let%Opt usd = usdJson;
       let%Opt btc = btcJson;
-      let%Opt result = decode(usd, btc);
+      let%Opt supply =
+        switch (supplyJson) {
+        | None => None
+        | Some(sp) => sp |> Js.Json.decodeNumber
+        };
+      let%Opt result = decode(usd, btc, supply);
```
Comment (evilpeach) [inline scan/src/rpc/PriceHook.re]: Use `Belt.Option.map` also.

File: scan/src/rpc/PriceHook.re [hunk 5]
```diff
@@ -5,12 +5,12 @@ type t = {
...
   btcMarketCap: float,
   btc24HrChange: float,
-  // circulatingSupply: float,
+  circulatingSupply: float,
 };
 
 module CoinGekco = {
   let get = () => {
-    let decode = (usdJson, btcJson) =>
+    let decode = (usdJson, btcJson, supplyJson) =>
```
Comment (evilpeach) [inline scan/src/rpc/PriceHook.re]: name it as `circulatingSupply`

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

**Comment by evilpeach** [comment]
@pzshine Please review this.

**Review by pzshine — APPROVED** [review]

**Pull request merged**